.azure-pipelines/generate-modules-template.yml

```diff
@@ -41,7 +41,22 @@ jobs:
 
   steps:
   - template: ./common-templates/install-tools-template.yml
+
+  - task: PowerShell@2
+    name: "ComputeBranch"
+    displayName: "Compute Docs Branch Name"
+    inputs:
+        targetType: inline
+        script: |
+            $currentDocsBranch = "DocsGeneration"
+            $branch = "{0}/{1}" -f $currentDocsBranch, (Get-Date -Format yyyyMMddHHmm)
+            Write-Host "Docs Branch $branch"
+            Write-Host "##vso[task.setvariable variable=DocsGenerationBranch;isOutput=true]$branch"
+
   - template: ./common-templates/checkout.yml
+    parameters:
+        TargetBranch: $(ComputeBranch.DocsGenerationBranch)
```

Comment:
This change locks the template used by other pipelines to the docs generation branch. This will break the current CI/CD pipeline and our weekly open API refresh pipeline.

Ideally, a template should be agnostic of a branch as other pipelines will reuse it.